Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Adaptive_sizing_field.h

```diff
@@ -0,0 +1,337 @@
+      calc_sizing_map(pmesh, np);
+    }
+    else
+    {
+      // expand face selection and calculate curvature from it
+      std::vector<face_descriptor> selection(face_range.begin(), face_range.end());
+      auto is_selected = get(CGAL::dynamic_face_property_t<bool>(), pmesh);
+      for (face_descriptor f : faces(pmesh)) put(is_selected, f, false);
+      for (face_descriptor f : face_range)  put(is_selected, f, true);
+      expand_face_selection(selection, pmesh, 1,
+                            is_selected, std::back_inserter(selection));
+      Face_filtered_graph<PolygonMesh> ffg(pmesh, selection);
+
+      calc_sizing_map(ffg, np);
+    }
+  }
+
+  /*!
+  * returns an object to serve as criteria for adaptive curvature-based edge lengths. It calls
+  * the first constructor using default values for the vertex point map of the input polygon mesh.
+  *
+  * @tparam FaceRange range of `boost::graph_traits<PolygonMesh>::%face_descriptor`,
+  *         model of `Range`. Its iterator type is `ForwardIterator`.
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param tol the error tolerance, used together with curvature to derive target edge length.
+  *        Lower tolerance values will result in shorter mesh edges.
+  * @param edge_len_min_max is the stopping criterion for minimum and maximum allowed
+  *        edge length.
+  * @param face_range the range of triangular faces defining one or several surface patches
+  *        to be remeshed. It should be the same as the range of faces used for `isotropic_remeshing()`.
+  * @param pmesh a polygon mesh with triangulated surface patches to be remeshed. It should be the
+  *              same mesh as the one used in `isotropic_remeshing()`. Additionally, the default vertex point
+  *              map of pmesh is used to construct the class.
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+
+  * \cgalNamedParamsBegin
+  *   \cgalParamNBegin{ball_radius}
+  *     \cgalParamDescription{a scalar value specifying the radius used for expanding curvature measures.
+  *                           It can potentially smooth the curvature and consequently the sizing field in the
+   *                          case of noisy input.}
+  *     \cgalParamDescription{a scalar value specifying the radius used for expanding curvature measures.
+  *                           It can effectively smooth the curvature field and consequently the sizing field.}
+  *     \cgalParamType{`Base::FT`}
+  *     \cgalParamDefault{`-1`}
+  *     \cgalParamExtra{If this parameter is omitted (`-1`), the expansion will just by a weightless sum of
+  *                     measures on faces around the vertex.}
+  *   \cgalParamNEnd
+  * \cgalNamedParamsEnd
+  */
+  template <typename FaceRange
+          , typename NamedParameters = parameters::Default_named_parameters>
+  Adaptive_sizing_field(const double tol
```

Comment:
The data member is of type `FT` and not double. Why this mismatch?

Reply:
Fixed